dvc/external_repo.py

```diff
@@ -62,16 +62,33 @@ def _external_repo(url=None, rev=None, cache_dir=None):
 
     # Adjust new clone/copy to fit rev and cache_dir
     repo = Repo(new_path)
+    original_repo = None
     try:
         if rev is not None:
             repo.scm.checkout(rev)
+        # check if the URL is local and no default remote
+        # add default remote pointing to the original repo's cache location
+        if not os.path.isdir(url):
+            # Adjust original repo for pointing remote towards its' cache
+            original_repo = Repo(url)
+            rconfig = RemoteConfig(original_repo.config)
+            if not _remote_config_exists(rconfig):
+                rconfig.add(
```

Comment:
You shouldn't modify original repo, you should modify our clone, same as we do with `cache_dir` down below.

Reply:
This is not resolved yet, please take a note the you are modyfing `original_repo` config, while you should modify temporary one.

Reply:
@pared `rconfig` is the remote config of the temporary repo and not the original one.

Reply:
@pared I think you're commenting on an older version it is fixed in the latest version. Sorry I didn't mark this conversation as resolved.